<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>804</width>
    <height>586</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>ReDuino</string>
  </property>
  <property name="windowIcon">
   <iconset>
    <normaloff>:/main</normaloff>:/main</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="margin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="1" column="0">
     <widget class="QTabWidget" name="tabWidget">
      <property name="font">
       <font>
        <family>MS Shell Dlg 2</family>
        <pointsize>10</pointsize>
        <weight>50</weight>
        <bold>false</bold>
       </font>
      </property>
      <property name="cursor">
       <cursorShape>ArrowCursor</cursorShape>
      </property>
      <property name="tabPosition">
       <enum>QTabWidget::North</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>1</number>
      </property>
      <property name="elideMode">
       <enum>Qt::ElideNone</enum>
      </property>
      <property name="documentMode">
       <bool>false</bool>
      </property>
      <property name="tabsClosable">
       <bool>true</bool>
      </property>
      <property name="movable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Start</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QPlainTextEdit" name="mainEdit">
          <property name="documentTitle">
           <string>start</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>New File</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6">
        <property name="margin">
         <number>0</number>
        </property>
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="1" column="0">
         <widget class="CodeEditor" name="mainEdit_2">
          <property name="font">
           <font>
            <family>Arial</family>
            <pointsize>10</pointsize>
            <weight>50</weight>
            <bold>false</bold>
            <stylestrategy>PreferAntialias</stylestrategy>
           </font>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Sunken</enum>
          </property>
          <property name="midLineWidth">
           <number>1</number>
          </property>
          <property name="documentTitle">
           <string/>
          </property>
          <property name="plainText">
           <string>// Project 12 - Melody Player

#define GAMMAOFFSET 4

int speakerPin = 12;
int bassPin = 11;

 boolean flag = true;
 int incomingByte = 0;


int freq[8][12]={ 
    {16, 17, 18, 19, 20, 21, 23, 24, 25, 27, 29, 30},
    {32, 34, 36, 38, 41, 43, 46, 49, 51, 55, 58, 61},
    {65, 69, 73, 77, 82, 87, 92, 98, 103, 110, 116, 123},
    {130, 138, 146, 155, 164, 174, 185, 196, 207, 220, 233, 246},
    {261, 277, 293, 311, 329, 349, 369, 392, 415, 440, 466, 493},
    {523, 554, 587, 622, 659, 698, 739, 783, 830, 880, 932, 987},
    {1046, 1108, 1174, 1244, 1318, 1396, 1479, 1567, 1661, 1760, 1864, 1975},
    {2093, 2217, 2349, 2489, 2637, 2793, 2959, 3135, 3322, 3520, 3729, 3951}
};

int tempo = 50;

void playTone(int tone, int durationtemp)
{  
      unsigned long duration=(unsigned long)((64*tempo)/durationtemp);

      double fuck=1000000.0/tone;

      unsigned long del=(unsigned long)(fuck);

      for (long i = 0; i &lt; duration * 1000L; i += del * 2) {
           if (tone==6000){
               digitalWrite(speakerPin, LOW);
               delayMicroseconds(del*2);
           } else
           {
            digitalWrite(speakerPin, HIGH);
            delayMicroseconds(del);
            digitalWrite(speakerPin, LOW);
            delayMicroseconds(del);
           }
      }
}

void setup() {

  pinMode(speakerPin, OUTPUT);
  pinMode(bassPin, OUTPUT);
  pinMode(13,OUTPUT);
  pinMode(4,OUTPUT);
  digitalWrite(4,HIGH);
  //digitalWrite(13,HIGH);
  analogWrite(13,1000);
  Serial.begin(115200);
  Serial.print(LOW);
  Serial.print('-');
  Serial.print(HIGH);
  attachInterrupt(0,inter,CHANGE);
  
}

void inter(){
    Serial.print(&quot;Interrupt!\n&quot;); 
}


int note(char c1, char c2)
{
  int r=0;
  switch (c2){
      case '#':
                 switch (c1){
                    case 'c': r=1; break;
                    case 'd': r=3; break;
                    case 'f': r=6; break;
                    case 'g': r=8; break;
                    case 'a': r=10; break;  
                    case 'P': r=200; break;
                    case 'C': r=1; break;
                    case 'D': r=3; break;
                    case 'F': r=6; break;
                    case 'G': r=8; break;
                    case 'A': r=10; break;  
                    default: break;          
                 }
              break;
              
      case '*':
         switch (c1){
            case 'c': r=0; break;
            case 'd': r=2; break;
            case 'e': r=4; break;
            case 'f': r=5; break;
            case 'g': r=7; break;
            case 'a': r=9; break;            
            case 'h': r=11; break;
            case 'P': r=200; break;
            case 'C': r=0; break;
            case 'D': r=2; break;
            case 'E': r=4; break;
            case 'F': r=5; break;
            case 'G': r=7; break;
            case 'A': r=9; break;            
            case 'H': r=11; break;
            default:break;
         }
             break;
      case 'b':
         switch (c1){
            case 'd': r=1; break;
            case 'e': r=3; break;
            case 'g': r=6; break;
            case 'a': r=8; break;
            case 'h': r=10; break;
            case 'D': r=1; break;
            case 'E': r=3; break;
            case 'G': r=6; break;
            case 'A': r=8; break;
            case 'H': r=10; break;
            case 'P': r=200; break;
            default:break;         
         }
      break;
  }
  return r;  
}

void pause(int p)
{
  playTone(6000, p);
  return;
}

void playPattern(char p[5])
{
  if(note(p[2], p[3]) == 200)
    pause( (p[0]-48)*10+(p[1]-48) );
  else
    playTone(freq[(p[4]-48)+GAMMAOFFSET][note(p[2],p[3])], (p[0]-48)*10+(p[1]-48));  
}

void play(char m[], int length)
{
  char *pattern;
  
  for(int i=0;i&lt;length;i+=6)
      playPattern(m+i);
}

char melody1[]=&quot;32c*0-32c#0-32d*0-32d#0-32e*0-32f*0-32f#0-32g*0-32g#0-32a*0-32a#0-32h*0&quot;;
char melody2[]=&quot;32h*5-32a#5-32a*5-32g#5-32g*5-01P*0-32f*5-32e*5-32d#5-32d*5-32c#5-32c*5&quot;;


/*
char NEM_1[] = &quot;08e*4-08g*5-08h*5-08e*6-08h*5-08g*5&quot;;
char NEM_1_2[] = &quot;08e*4-08g*5-08h*5-08e*6-08h*5-08h*6&quot;;
char NEM_2[] = &quot;08h*6-08g*5-08h*5-16h*6-16e*5-08h*5-08g*5&quot;;
char NEM_3[] = &quot;&quot;;*/

char LEDZEPPELIN[] = &quot;08c*2-16P*0-08e*2-08f#2-16a*2-08g*2-16P*0-08e*2-08c*2-16a*1-08f#1-16P*0-02g*1-04P*0-08c*2-16P*0-08e*2-08f#2-16e*3-08d*3-16P*0-08c*3-08a#2-16g*2-08d*2-16P*0-02e*2-&quot;;
void loop() 
{
 /*  play(NEM_1,36);
   play(NEM_1,36);
   play(NEM_1,36);
   play(NEM_1_2,36);*/
   
 //  play(LEDZEPPELIN,26*6);
 int d = 1000;
 int octave = 4;
 
if(flag==true)
{
   noTone(speakerPin);
   tone(bassPin,freq[octave][7],5000);

   delay(d);
   noTone(bassPin);
   tone(speakerPin,freq[octave][0],5000);
   delay(d);
  
   Serial.print(&quot;Hello, fags!\n&quot;);
}

   if (Serial.available() &gt; 0) {
		// read the incoming byte:
		incomingByte = Serial.read();

		// say what you got:
		Serial.print(&quot;I received. &quot;);		
                Serial.println(incomingByte, DEC);
                
                if(incomingByte == 'g')
                {
                  Serial.print(&quot;GOGOGOGO!!!\n&quot;);
                  flag = true;
                }
                else
                {
                  if(incomingByte == 's')
                  {
                    Serial.print(&quot;STOP BIATCH!\n&quot;);
                    flag = false;
                  }
                }

	} 

}</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="projectDock">
   <property name="minimumSize">
    <size>
     <width>274</width>
     <height>146</height>
    </size>
   </property>
   <property name="maximumSize">
    <size>
     <width>274</width>
     <height>524287</height>
    </size>
   </property>
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Project</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <property name="styleSheet">
     <string notr="true">margin:0px;
padding:0px;</string>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="spacing">
      <number>0</number>
     </property>
     <property name="margin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTreeWidget" name="projectTree">
       <property name="font">
        <font>
         <pointsize>10</pointsize>
         <stylestrategy>PreferAntialias</stylestrategy>
        </font>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="styleSheet">
        <string notr="true">border: 2px solid darkgray;</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Sunken</enum>
       </property>
       <property name="lineWidth">
        <number>1</number>
       </property>
       <property name="midLineWidth">
        <number>1</number>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAsNeeded</enum>
       </property>
       <property name="tabKeyNavigation">
        <bool>true</bool>
       </property>
       <property name="showDropIndicator" stdset="0">
        <bool>true</bool>
       </property>
       <property name="rootIsDecorated">
        <bool>false</bool>
       </property>
       <property name="sortingEnabled">
        <bool>false</bool>
       </property>
       <property name="animated">
        <bool>true</bool>
       </property>
       <property name="headerHidden">
        <bool>true</bool>
       </property>
       <property name="expandsOnDoubleClick">
        <bool>true</bool>
       </property>
       <property name="columnCount">
        <number>1</number>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="headerCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="headerShowSortIndicator" stdset="0">
        <bool>false</bool>
       </attribute>
       <attribute name="headerCascadingSectionResizes">
        <bool>true</bool>
       </attribute>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="headerShowSortIndicator" stdset="0">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string notr="true">Файл</string>
        </property>
       </column>
       <item>
        <property name="text">
         <string>Project14</string>
        </property>
        <item>
         <property name="text">
          <string>main.pde</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sources</string>
         </property>
         <item>
          <property name="text">
           <string>serial.pde</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>motor.pde</string>
          </property>
         </item>
        </item>
        <item>
         <property name="text">
          <string>Headers</string>
         </property>
         <item>
          <property name="text">
           <string>tools.h</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>lib.h</string>
          </property>
         </item>
        </item>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="logDock">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>112</width>
     <height>132</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="windowTitle">
    <string>Output</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>8</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QGridLayout" name="gridLayout_2">
     <property name="margin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QTextBrowser" name="logArea">
       <property name="styleSheet">
        <string notr="true">border: 2px solid darkgray;</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="funcDock">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="minimumSize">
    <size>
     <width>140</width>
     <height>132</height>
    </size>
   </property>
   <property name="baseSize">
    <size>
     <width>200</width>
     <height>0</height>
    </size>
   </property>
   <property name="floating">
    <bool>false</bool>
   </property>
   <property name="features">
    <set>QDockWidget::AllDockWidgetFeatures</set>
   </property>
   <property name="windowTitle">
    <string>Functions</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_3">
    <layout class="QGridLayout" name="gridLayout_4">
     <property name="margin">
      <number>0</number>
     </property>
     <property name="spacing">
      <number>0</number>
     </property>
     <item row="0" column="0">
      <widget class="QTreeWidget" name="funcTree">
       <property name="font">
        <font>
         <stylestrategy>PreferAntialias</stylestrategy>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">border: 2px solid darkgray;</string>
       </property>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
       <attribute name="headerVisible">
        <bool>false</bool>
       </attribute>
       <column>
        <property name="text">
         <string>Функция</string>
        </property>
       </column>
       <item>
        <property name="text">
         <string>void playTone(int tone, int durationtemp)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>void setup()</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>void inter()</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>int note(char c1, char c2)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>void pause(int p)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>void playPattern(char p[5])</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>void play(char m[], int length)</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>void loop()</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>804</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>&amp;Edit</string>
    </property>
    <addaction name="actionUndo"/>
    <addaction name="actionRedo"/>
    <addaction name="separator"/>
    <addaction name="actionCopy"/>
    <addaction name="actionPaste"/>
    <addaction name="actionDelete"/>
    <addaction name="separator"/>
    <addaction name="actionSelectAll"/>
    <addaction name="separator"/>
    <addaction name="actionFindReplace"/>
    <addaction name="actionFind"/>
    <addaction name="separator"/>
    <addaction name="actionAutoFormat"/>
    <addaction name="separator"/>
    <addaction name="actionComment"/>
    <addaction name="actionUncomment"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>&amp;View</string>
    </property>
   </widget>
   <widget class="QMenu" name="menu_5">
    <property name="title">
     <string>&amp;Window</string>
    </property>
    <widget class="QMenu" name="menuPanels">
     <property name="title">
      <string>Panels</string>
     </property>
     <addaction name="actionProject_Dock"/>
     <addaction name="actionFunctions_Dock"/>
     <addaction name="actionOutput_Dock"/>
     <addaction name="actionToolbar"/>
    </widget>
    <addaction name="actionFull_Screen"/>
    <addaction name="menuPanels"/>
   </widget>
   <widget class="QMenu" name="menu_6">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout_this_program"/>
    <addaction name="actionAbout_At"/>
   </widget>
   <widget class="QMenu" name="menu_4">
    <property name="title">
     <string>&amp;Project</string>
    </property>
    <widget class="QMenu" name="menuBoard">
     <property name="title">
      <string>Board</string>
     </property>
     <addaction name="actionType"/>
     <addaction name="actionConnection"/>
    </widget>
    <addaction name="actionCompile"/>
    <addaction name="actionUpload"/>
    <addaction name="actionCompileUpload"/>
    <addaction name="separator"/>
    <addaction name="menuBoard"/>
   </widget>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <widget class="QMenu" name="menu_Open">
     <property name="title">
      <string>&amp;Open</string>
     </property>
     <addaction name="actionProject_2"/>
     <addaction name="actionFile_2"/>
    </widget>
    <widget class="QMenu" name="menuNew">
     <property name="title">
      <string>&amp;New</string>
     </property>
     <addaction name="actionProject"/>
     <addaction name="actionFile"/>
    </widget>
    <widget class="QMenu" name="menuOpen_Recent">
     <property name="title">
      <string>Open &amp;Recent</string>
     </property>
     <addaction name="action1_pde"/>
     <addaction name="actionSosat_pde"/>
    </widget>
    <addaction name="menuNew"/>
    <addaction name="separator"/>
    <addaction name="menu_Open"/>
    <addaction name="menuOpen_Recent"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
   <addaction name="menu_4"/>
   <addaction name="menu_5"/>
   <addaction name="menu_6"/>
  </widget>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>50</width>
     <height>30</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpenIcon"/>
   <addaction name="actionSaveIcon"/>
   <addaction name="separator"/>
   <addaction name="actionUndoIcon"/>
   <addaction name="action_Redo"/>
   <addaction name="separator"/>
   <addaction name="actionCompileIcon"/>
   <addaction name="actionUploadIcon"/>
   <addaction name="separator"/>
   <addaction name="actionSerialMonitor"/>
  </widget>
  <action name="actionExit">
   <property name="text">
    <string>&amp;Exit</string>
   </property>
  </action>
  <action name="actionOpenIcon">
   <property name="icon">
    <iconset>
     <normaloff>:/open</normaloff>:/open</iconset>
   </property>
   <property name="text">
    <string>&amp;Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSaveIcon">
   <property name="icon">
    <iconset>
     <normaloff>:/save</normaloff>:/save</iconset>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>&amp;Save</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save &amp;As</string>
   </property>
  </action>
  <action name="actionUndo">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="icon">
    <iconset>
     <normaloff>:/first-png</normaloff>:/first-png</iconset>
   </property>
   <property name="text">
    <string>Undo</string>
   </property>
  </action>
  <action name="actionPaste">
   <property name="text">
    <string>&amp;Paste</string>
   </property>
  </action>
  <action name="actionUndoIcon">
   <property name="icon">
    <iconset>
     <normaloff>:/undo</normaloff>:/undo</iconset>
   </property>
   <property name="text">
    <string>&amp;Undo</string>
   </property>
  </action>
  <action name="action_Redo">
   <property name="icon">
    <iconset>
     <normaloff>:/redo</normaloff>:/redo</iconset>
   </property>
   <property name="text">
    <string>&amp;Redo</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="text">
    <string>&amp;Delete</string>
   </property>
  </action>
  <action name="actionSelectAll">
   <property name="text">
    <string>&amp;Select All</string>
   </property>
  </action>
  <action name="actionFind">
   <property name="text">
    <string>&amp;Find</string>
   </property>
  </action>
  <action name="actionFindReplace">
   <property name="text">
    <string>Find Replac&amp;e</string>
   </property>
  </action>
  <action name="actionCompileIcon">
   <property name="icon">
    <iconset>
     <normaloff>:/compile</normaloff>:/compile</iconset>
   </property>
   <property name="text">
    <string>&amp;Compile</string>
   </property>
  </action>
  <action name="actionUploadIcon">
   <property name="icon">
    <iconset>
     <normaloff>:/upload</normaloff>:/upload</iconset>
   </property>
   <property name="text">
    <string>&amp;Upload</string>
   </property>
  </action>
  <action name="actionCompileUpload">
   <property name="text">
    <string>C&amp;ompile&amp;&amp;Upload</string>
   </property>
  </action>
  <action name="actionConnection">
   <property name="text">
    <string>Connection</string>
   </property>
  </action>
  <action name="actionSerialMonitor">
   <property name="icon">
    <iconset>
     <normaloff>:/monitor</normaloff>:/monitor</iconset>
   </property>
   <property name="text">
    <string>Serial Monitor</string>
   </property>
  </action>
  <action name="actionType">
   <property name="text">
    <string>Type</string>
   </property>
  </action>
  <action name="actionAutoFormat">
   <property name="text">
    <string>Auto Format</string>
   </property>
  </action>
  <action name="actionProject">
   <property name="text">
    <string>&amp;Project</string>
   </property>
  </action>
  <action name="actionFile">
   <property name="text">
    <string>&amp;File</string>
   </property>
  </action>
  <action name="actionProject_2">
   <property name="text">
    <string>&amp;Project</string>
   </property>
  </action>
  <action name="actionFile_2">
   <property name="text">
    <string>&amp;File</string>
   </property>
  </action>
  <action name="action1_pde">
   <property name="text">
    <string>1.pde</string>
   </property>
  </action>
  <action name="actionSosat_pde">
   <property name="text">
    <string>Sosat.pde</string>
   </property>
  </action>
  <action name="actionComment">
   <property name="text">
    <string>Comment</string>
   </property>
  </action>
  <action name="actionUncomment">
   <property name="text">
    <string>Uncomment</string>
   </property>
  </action>
  <action name="actionFull_Screen">
   <property name="text">
    <string>Full Screen</string>
   </property>
  </action>
  <action name="actionAbout_this_program">
   <property name="text">
    <string>About Reeno</string>
   </property>
  </action>
  <action name="actionAbout_At">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionProject_Dock">
   <property name="text">
    <string>Project Dock</string>
   </property>
  </action>
  <action name="actionFunctions_Dock">
   <property name="text">
    <string>Functions Dock</string>
   </property>
  </action>
  <action name="actionOutput_Dock">
   <property name="text">
    <string>Output Dock</string>
   </property>
  </action>
  <action name="actionToolbar">
   <property name="text">
    <string>Toolbar</string>
   </property>
  </action>
  <action name="actionCompile">
   <property name="text">
    <string>Compile</string>
   </property>
  </action>
  <action name="actionUpload">
   <property name="text">
    <string>Upload</string>
   </property>
  </action>
  <action name="actionRedo">
   <property name="text">
    <string>Redo</string>
   </property>
  </action>
  <action name="actionCopy">
   <property name="text">
    <string>Copy</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>CodeEditor</class>
   <extends>QPlainTextEdit</extends>
   <header location="global">codeditor.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
